core/src/test/java/org/testcontainers/utility/DockerImageNameTest.java

```diff
@@ -0,0 +1,18 @@
+package org.testcontainers.utility;
+
+import org.junit.Test;
+
+public class DockerImageNameTest {
+
+    @Test
+    public void validNames() {
+        DockerImageName.validate("myname:latest"); // no repo
+        DockerImageName.validate("repo/my-name:1.0"); // no repo
+        DockerImageName.validate("repo.foo.com:1234/my-name:1.0"); // no repo
+    }
```

Comment:
While testing the code on our private repo (it worked), I managed to fell through some cracks which revealed this class: NameParser contained in docker-java library. It has parser for tags which we can use instead of the class I wrote DockerImageName. I'm thinking of removing it and using NameParser.parseRepositoryTag(). 
The thing is if you follow command trailer from GenericContainer.setImageName() you get eventually to NameParser.parseRepositoryTag(). So calling it before seems a bit redundant. 
The only reason to do this is if we want to force a tag. Their method simply declares the tag as empty in case it doesn't parse well (contain /), and of course it doesn't fail on missing tag.

WDYT?